packages/shared/src/hooks/useSharePost.ts

```diff
@@ -1,101 +1,80 @@
-import { useContext, useMemo, useState } from 'react';
+import { useCallback, useContext } from 'react';
 import { Post } from '../graphql/posts';
-import { FeedItemPosition, postAnalyticsEvent } from '../lib/feed';
+import { postAnalyticsEvent } from '../lib/feed';
 import AnalyticsContext from '../contexts/AnalyticsContext';
 import { ShareProvider } from '../lib/share';
 import { Origin } from '../lib/analytics';
 import { useCopyPostLink } from './useCopyPostLink';
 import { useGetShortUrl } from './utils/useGetShortUrl';
-import { ReferralCampaignKey } from '../lib/referral';
+import { ReferralCampaignKey } from '../lib';
+import { useLazyModal } from './useLazyModal';
+import { LazyModal } from '../components/modals/common/types';
+import { ShareModalProps } from '../components/modals/post/common';
 
-export function useSharePost(origin: Origin): {
-  sharePost: Post;
-  sharePostFeedLocation?: FeedItemPosition;
-  openSharePost: (Post, columns?, column?, row?) => void;
-  copyLink: (Post, columns?, column?, row?) => void;
-  closeSharePost: () => void;
-  openNativeSharePost?: (Post, columns?, column?, row?) => void;
-} {
+type FuncProps = Omit<ShareModalProps, 'origin'>;
+
+interface UseSharePost {
+  openSharePost: (props: FuncProps) => void;
+  copyLink: (props: FuncProps) => void;
+  openNativeSharePost?: (post: Post) => void;
+}
+
+export function useSharePost(origin: Origin): UseSharePost {
   const { trackEvent } = useContext(AnalyticsContext);
-  const [shareModal, setShareModal] = useState<Post>();
   const [, copyLink] = useCopyPostLink();
-  const [sharePostFeedLocation, setSharePostFeedLocation] =
-    useState<FeedItemPosition>({});
   const { getShortUrl, getTrackedUrl } = useGetShortUrl();
+  const { openModal } = useLazyModal();
```

Comment:
Most changes here are related to migrating to `useLazyModal` over using two states to manage the modal.